test/internal/connector.test.js

```diff
@@ -140,7 +141,7 @@ describe('connector', () => {
 
   it('should convert failure messages to errors', done => {
     const channel = new DummyChannel.channel;
-    connection = new Connection(channel, 'bolt://localhost');
+    connection = new Connection(channel, 'bolt://localhost', Logger.noOp());
```

Comment:
What if the logger argument isn't defined, does it break?

Reply:
Yes, this test breaks. This is expected, right?